pkg/image/strategy/simplelookup.go

```diff
@@ -0,0 +1,159 @@
+package strategy
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	operatorv1alpha1scheme "github.com/openshift/client-go/operator/clientset/versioned/scheme"
+	operatorv1alpha1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1alpha1"
+	reference "github.com/openshift/library-go/pkg/image/reference"
+	"github.com/openshift/library-go/pkg/image/registryclient"
+)
+
+// simpleLookupICSP holds ImageContentSourcePolicy variables to look up image sources
+// satisfies *Context AlternativeRepositoryStrategy interface
+type simpleLookupICSP struct {
+	lock sync.Mutex
+
+	alternates []reference.DockerImageReference
+	icspFile   string
+	icspClient operatorv1alpha1client.ImageContentSourcePolicyInterface
```

Comment:
This should be in a separate package as discussed before

Reply:
cleaned up commits & cherry-picks